src/ossl/ecdsa.rs

```diff
@@ -193,6 +199,9 @@ impl EcdsaOperation {
             CKM_ECDSA_SHA3_384,
             CKM_ECDSA_SHA3_512,
         ] {
+            if cfg!(feature = "no_sha1") && ckm == &CKM_ECDSA_SHA1 {
+                continue;
+            }
```

Comment:
You should also do this in the RSA, HMAC, ECDH, ... where the sha1 based mechanisms are registered. They contribute to the list of supported mechanisms that is returned to the user.

Reply:
This comment brings up a blind spot in this PR, namely no negative tests to ensure it is working.

Reply:
Could we use there something like this too?
```rust
        let list = [
            CKM_ECDSA,
            #[cfg(not(feature = "no_sha1"))]
            CKM_ECDSA_SHA1,
            CKM_ECDSA_SHA224,
            CKM_ECDSA_SHA256,
            CKM_ECDSA_SHA384,
            CKM_ECDSA_SHA512,
            CKM_ECDSA_SHA3_224,
            CKM_ECDSA_SHA3_256,
            CKM_ECDSA_SHA3_384,
            CKM_ECDSA_SHA3_512,
        ]
        for ckm in list {
```

Reply:
Thanks, noted